### ISSUE TITLE
Fixing paltest_raiseexception_test3 on ARM

### ISSUE THREAD

**Issue by manu-silicon** [dotnet/coreclr#3898]
The test suite for PAL on ARM is almost successful:

```
The following test(s) failed:
exception_handling/RaiseException/test3/paltest_raiseexception_test3. Exit code: 139

PAL Test Results:
  Passed: 808
  Failed: 1
```

The test that is failing is ```exception_handling/RaiseException/test3/paltest_raiseexception_test3```.

**Comment by manu-silicon** [comment]
gdb call stack when it fails:

```
(gdb) where
#0  0xb6688a62 in ?? () from /usr/lib/arm-linux-gnueabihf/libunwind-arm.so.8
#1  0xb66894ea in _Uarm_step () from /usr/lib/arm-linux-gnueabihf/libunwind-arm.so.8
#2  0x000400c2 in PAL_VirtualUnwind (context=0xbe7ff678, contextPointers=0x0) at /home/ubuntu/local/Microsoft/coreclr/src/pal/src/exception/seh-unwind.cpp:277
#3  0x00040672 in RtlpRaiseException (ExceptionRecord=0xbe7ff874) at /home/ubuntu/local/Microsoft/coreclr/src/pal/src/exception/seh-unwind.cpp:591
#4  0x000405de in RaiseException (dwExceptionCode=3221225612, dwExceptionFlags=0, nNumberOfArguments=0, lpArguments=0x0)
    at /home/ubuntu/local/Microsoft/coreclr/src/pal/src/exception/seh-unwind.cpp:655
#5  0x000262ac in main::$_1::operator() (this=0xbe7ff97c, unused=0x0) at /home/ubuntu/local/Microsoft/coreclr/src/pal/tests/palsuite/exception_handling/RaiseException/test3/test.cpp:69
#6  0x000260fe in main (argc=1, argv=0xbe7ffae4) at /home/ubuntu/local/Microsoft/coreclr/src/pal/tests/palsuite/exception_handling/RaiseException/test3/test.cpp:76
```

**Comment by manu-silicon** [comment]
@myungjoo Although there is no managed code here, could this be related to your issue #3859?

**Comment by myungjoo** [comment]
This does not look like to be related with #3859, but you may need to ensure if CONTEXT_CaptureContext() in RtlpRaiseException() is filling ContextRecord correctly, especially for the values required by WinContextToUnwindCursor() in seh-unwind.cpp.

**Comment by manu-silicon** [comment]
If I set ```UNW_ARM_UNWIND_METHOD``` to 1, then the test passes. It is in conflict with fix made at commit https://github.com/dotnet/coreclr/commit/35b5df6888f1574e6ef9313c6bc8522d1817573d.

**Comment by myungjoo** [comment]
If you are getting segmentation faults with the case, could you please test with the libunwind-arm patched with?:
```
diff --git a/src/arm/Ginit.c b/src/arm/Ginit.c
index 1ed3dbf..c643032 100644
--- a/src/arm/Ginit.c
+++ b/src/arm/Ginit.c
@@ -128,6 +128,11 @@ access_mem (unw_addr_space_t as, unw_word_t addr, unw_word_t *val, int write,
 {
   if (write)
     {
+      /* validate address */
+      const struct cursor *c = (const struct cursor *) arg;
+      if (c && validate_mem(addr))
+        return -1;
+
       Debug (16, "mem[%x] <- %x\n", addr, *val);
       *(unw_word_t *) addr = *val;
     }
```

Besides, installing libunwind-dbg will be helpful as well (showing where it has died within libunwind)

Note that Android-libunwind is modified similarly as this patch and some Linux distro (probably centos or fedora or so, which uses rpmbuild and spec files) prohibits UNW-ARM DWARF mode because of its instability.

**Comment by manu-silicon** [comment]
@myungjoo I'm already using the gdb version for debugging, it was failing when calling ```unw_step``` in the parth that exercises the ```UNW_ARM_METHOD_FRAME``` condition at line 233 of arm/Gstep.c. The issue is that the ```frame``` variable had the content of one of the register and not a valid address, and it would fail in ```access_mem``` like you mentioned, but on the read side, not the write one.

I tried to compile your code but ```validate_mem``` is not present for me on arm for the source code I got coming from ubuntu.

I found the code directly from the savanah git repository for libunwind and they do have a protection on read for access_mem. I guess Ubuntu 14.04 includes a relatively old version.

Using that version solved this problem.

Should CoreCLR includes the proper version?
